pkg/events/recorder/recorder.go

```diff
@@ -67,7 +68,12 @@ func (sr *SimpleRecorder) makeEvent(ref *v1.ObjectReference, labels map[string]s
 	t := metav1.NewTime(now)
 	namespace := ref.Namespace
 	if namespace == "" {
-		namespace = metav1.NamespaceDefault
+		if key := os.Getenv("NAMESPACE"); key != "" {
```

Comment:
CSI generates events for cluster scoped CRs like `drives` and namespaced CRs like `volumes`.  How will that work when CSI in installed in non-default namespace, but volume CR is in default?

Reply:
I think we need to add a check here, it is for drives only.

Reply:
updated

Reply:
Sorry, my comment was wrong. For cluster scoped CRs namespace will be empty, for namespaced - non-empty. @safronovD am I correct?

Reply:
ref.Nmespace here is empty or "default" for the namespaced CR while it is in default namespace?

Reply:
In my opinion, the first implementation was correct - `namespace == ""` means that resource is cluster-scoped, so we generate event for it in CSI NS and it doesn't matter is it `Drive` or something else